### PR TITLE
Added alerts for available starport missions and local maps

### PR DESCRIPTION
**Pull request by clintonmead**
As discussed in [issue 3620](https://github.com/endless-sky/endless-sky/issues/3620), here's a pull request implementing the following alerts (although the user must still take the actual action):

1. Alert when there is a mission waiting in the starport. (preference.txt option "Spaceport mission alert")
2. Alert when there is new local map information available in outfitting (preference.txt option "Local map available alert")

Note that the alerts to buy maps have the distance hard coded, instead of reading from the data file. As couldn't work out how to read it from the data file.

My pull request also has a bunch of whitespace changes, I couldn't entice my editor not to include them annoyingly.

### CODE REVIEW

**Comment by jafdy** [comment]
The first (`Spaceport mission alert`) has been noed it the past.

**Comment by Amazinite** [comment]
>The first (Spaceport mission alert) has been noed it the past.

Reference: #1292

File: source/PlayerInfo.cpp [hunk 1]
```diff
@@ -991,22 +995,45 @@ void PlayerInfo::Land(UI *ui)
...
+
+  // Alert when there's spaceport missions available
+  if (Preferences::Has("Spaceport mission alert") && hasSpaceport && MissionToOffer(Mission::SPACEPORT))
+  {
+    Messages::Add("There is a spaceport mission available.");
+    ui->Push(new Dialog("There is a spaceport mission available."));
+  }
+
+  // Alert when you can buy a map
+  if (Preferences::Has("Local map available alert") && hasOutfitter)
```
Comment (tehhowch) [inline source/PlayerInfo.cpp]: `if(Preferences.....`

File: source/PlayerInfo.cpp [hunk 2]
```diff
@@ -991,22 +995,45 @@ void PlayerInfo::Land(UI *ui)
...
+  {
+    // Hard coded distance, this is ugly.
```
Comment (tehhowch) [inline source/PlayerInfo.cpp]: You've used spaces and not tabs.

Comment (clintonmead) [reply source/PlayerInfo.cpp]: The whole point of this is to display on the main screen so you don't have to mindlessly click the outfitter. 

It's actually probably a good idea to do an ammo alert as well. 

I think however the approach pull request [#1292](https://github.com/endless-sky/endless-sky/pull/1292) takes is better though, with the red exclamation point on the starport.

I'd like to the same for the outfitter with local maps, and as you mentioned, it's probably worth adding ammo refills. But I need this info before entering the outfitter, that's the whole point.

Unfortunately, as far as I can tell, it seems the outfitting logic, like "CanBuy()", is tightly coupled with the outfitting GUI, and you can't use the logic from outside the GUI. Although perhaps I've missed something.

**Comment by clintonmead** [comment]
I'm closing this pull request because it seems the spaceport alert is only working for mainline missions anyway, not random missions.

I think the approach used in [#1292](https://github.com/endless-sky/endless-sky/pull/1292) is better. I'll try that approach, with both the spaceport and map info, and perhaps even a reloading alert. But I'll make a new PR for that.

**Comment by mootootwo** [comment]
I like that you have made it configurable in preferences.  If the spaceport alert defaults to off, it will be more likely to pass approval where 1292 failed.

**Pull request closed without merging**